### PR TITLE
Test basel/tensorflow with Java 17

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by iarspider** [comment]
please test for CMSSW_14_1_TF_X

**Comment by smuzaffar** [comment]
thanks @iarspider , did you try it for TF2.12/Bazel 5 which is in our default IBs?

File: java-env.spec [hunk 1]
```diff
@@ -8,7 +8,7 @@
 %if 0%{?rhel:1}
 %if 0%{?rhel} < 9
 # /usr/lib/jvm/java on EL8 contains java 1.8, and Bazel requires java 11
-%define java_home /usr/lib/jvm/java-11
+%define java_home /usr/lib/jvm/java-17
```
Comment (smuzaffar) [inline java-env.spec:11]: Note that this will break for slc7 as java jdk17 is not available for slc7

File: bazel.spec [hunk 2]
```diff
@@ -28,8 +28,12 @@ Patch1: bazel-absl
...
 %build
 
-export EXTRA_BAZEL_ARGS="--host_javabase=@local_jdk//:jdk --jobs %{compiling_processes}"
-export BAZEL_CXXOPTS="-Wno-error=deprecated:-std=c++%{cms_cxx_standard}"
+# Workaround for compilation issues in upb
+%define cms_cxx_standard 17
+
+export EXTRA_BAZEL_ARGS="--define=ABSOLUTE_JAVABASE=${JAVA_ROOT} --jobs %{compiling_processes}"
+export BAZEL_CXXOPTS="-std=c++%{cms_cxx_standard}"
+export JNI_FLAGS="--add-opens=java.base/java.nio=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED"
```
Comment (smuzaffar) [inline bazel.spec:36]: may be this should be done for java17 only? i.e. if java version in 17 or above then set this?

**Comment by iarspider** [comment]
No, but I can try. Don't we plan to update TF soon-ish?

**Comment by smuzaffar** [comment]
IIRC, TF 2.15 was failing for aarch64 that is why we did not move to it but you can re-try

**Comment by iarspider** [comment]
> IIRC, TF 2.15 was failing for aarch64 that is why we did not move to it but you can re-try

https://github.com/cms-sw/cmsdist/pull/9317

File: bazel.spec [hunk 3]
```diff
@@ -28,8 +28,12 @@ Patch1: bazel-absl
 
 %build
 
-export EXTRA_BAZEL_ARGS="--host_javabase=@local_jdk//:jdk --jobs %{compiling_processes}"
-export BAZEL_CXXOPTS="-Wno-error=deprecated:-std=c++%{cms_cxx_standard}"
+# Workaround for compilation issues in upb
+%define cms_cxx_standard 17
+
+export EXTRA_BAZEL_ARGS="--define=ABSOLUTE_JAVABASE=${JAVA_ROOT} --jobs %{compiling_processes}"
```
Comment (smuzaffar) [inline bazel.spec:34]: Should it be ${JAVA_HOME}?

**Comment by smuzaffar** [comment]
we can close this in favor of https://github.com/cms-sw/cmsdist/pull/9319

**Pull request closed without merging**